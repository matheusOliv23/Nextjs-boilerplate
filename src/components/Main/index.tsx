import React from "react"

export default function index() {
  return <h2>React avan√ßado</h2>
}
