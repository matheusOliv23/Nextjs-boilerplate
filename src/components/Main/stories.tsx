import React from "react"
import { ComponentStory, ComponentMeta } from "@storybook/react"

import Main from "./index"

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Main",
  component: Main,
  args: {
    title: "Boilerplate",
    description: "Typescript, Reactjs, Nextjs, Styled Components"
  },
  argTypes: {
    backgroundColor: { control: "color" }
  }
} as ComponentMeta<typeof Main>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Main> = (args) => <Main {...args} />

export const Basic = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Basic.args = {
  title: "Boilerplate",
  description: "Typescript, Reactjs, Nextjs, Styled Components"
}
