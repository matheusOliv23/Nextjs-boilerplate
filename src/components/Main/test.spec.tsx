import { render, screen } from "@testing-library/react"
import Main from "./index"

describe("Main", () => {
  it("should render heading", () => {
    render(<Main />)
    expect(
      screen.getByRole("heading", { name: /react avanÃ§ado/i })
    ).toBeInTheDocument()
  })

  it("should snapshot", () => {
    const { container } = render(<Main />)

    expect(container).toMatchSnapshot()
  })
})
