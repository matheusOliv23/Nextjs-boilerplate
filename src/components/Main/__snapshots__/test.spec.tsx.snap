// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should snapshot 1`] = `
<div>
  <h2>
    React avan√ßado
  </h2>
</div>
`;
